clients/src/test/java/org/apache/kafka/common/network/SslSelectorTest.java

```diff
@@ -86,13 +86,14 @@ public void testDisconnectWithIntermediateBufferedBytes() throws Exception {
         connect(node, new InetSocketAddress("localhost", server.port));
         String request = TestUtils.randomString(requestSize);
         selector.send(createSend(node, request));
+        assertTrue(selector.completedReceives().isEmpty());
 
         TestUtils.waitForCondition(new TestCondition() {
             @Override
             public boolean conditionMet() {
                 try {
                     selector.poll(0L);
-                    return selector.channel(node).hasBytesBuffered();
+                    return selector.channel(node).hasBytesBuffered() || !selector.completedReceives().isEmpty();
```

Comment:
This somewhat defeats the point of the test. We are trying to reach the state where `hasBytesBuffered` is set so that we can verify the removal of the selection key from `Selector.keysWithBufferedRead` on close. With this patch, it seems a bit tougher to enter this state.

Reply:
@hachikuji Yeah I agree with your concern. I don't have a very good solution on top of my head yet . Do you have any suggestion?

On the other hand this test is still useful after this change. Because without changing this line in the test, the test can still pass on my laptop with this patch. So this test can still work in some cases.